### ISSUE TITLE
Support NetStandard 1.3

### ISSUE THREAD

**Issue by NimaAra** [rpgmaker/NetJSON#144]
I understand you originally had _netstandard1.3_ support but after https://github.com/rpgmaker/NetJSON/issues/136, you upgraded to _netstandard1.6_; As you had already mentioned in that issue, _netstandard1.3_ is indeed supported by _netstandard1.6_ so I believe the issue was how the library was packaged up by _NuGet_, I am confident implementing  https://github.com/rpgmaker/NetJSON/issues/143 will address that issue.

Having the lowest _NetStandard_ support is important for me as I am planning on taking a dependency on _NetJSON_ in one of my libraries which I am planning on releasing soon but I do not want to bump up my version from _1.3_ to _1.6_.

**Comment by rpgmaker** [comment]
I can create another package today for just 1.3 I suppose.

Thanks,

**Comment by NimaAra** [comment]
There is no rush but what I am saying is that you shouldn't have to create a separate package for every version of dotnet. As long as you support the lowest common denominator you will be fine.

**Comment by rpgmaker** [comment]
I can try using the LCD as long as it works with standard 1.6.

Thanks,

**Comment by NimaAra** [comment]
Yes, if the minimum you can go to is 1.3, then it sure will be supported by anything higher. That is the whole idea behind the _netstandard_

**Comment by rpgmaker** [comment]
Does this commit help? https://github.com/rpgmaker/NetJSON/commit/4034b2e51f695eef29631e8d7f6700116ad07369 

If it does, then it is already on nuget and you can close the related issues

Thanks,

**Comment by NimaAra** [comment]
It does not look right, looking at the NuGet package I cannot find a _netstandard1.3_ also I can see too many frameworks with some even missing versions:

![Imgur](http://i.imgur.com/iyXOYyn.jpg)

Ideally, all you need is _net35_ and _netstandard1.3_ something like:

![Imgur](http://i.imgur.com/94jXKVh.jpg)

Note that in my case I am supporting _netstanard1.0_ but if _netstandard1.3_ is the lowest you can go then yours will have that instead of _netstandard1.3_

This is of course assuming that you are not doing anything special for a specific framework, e.g. doing something in _net45_ that you are not doing in _net35_ which I do not think you are.

So in summary having _net35_ and _netstandard1.3_ covers you for all the platforms that you are currently on.

**Comment by rpgmaker** [comment]
I need to create the folder for it when i deploy the nuget packages. The dll is compiled as netstandard 1.3

Sorry I am just replying now, I did not see the message until just now for some reason.

Thanks.

**Comment by rpgmaker** [comment]
I just realized that I did not change the proper reference to support 1.3, i was only updating the unit test. It seem i will have to do some refactoring to support a lower version from 1.6

**Comment by rpgmaker** [comment]
If you do have a PR that you can submit for the changes that will be great. If not, I can try to see what I can do to make 1.3 support the current reflection method such as Assembly.GetEntryAssembly() and Type#GetTypeInfo() and e.t.c

Thanks,

**Comment by NimaAra** [comment]
I will put together a PR hopefully this weekend. 

With regards to `Assembly.GetEntryAssembly()` this is only supported from _netstandard1.5_ I have not looked at the code to see how _NETJson_ is using this method but I can think of a couple of ways to tackle this, one would be to pass in the entry assembly via `Assembly.GetExecutingAssembly()` or the other option is to use `typeof(SomeType).GetTypeInfo().Asssembly` where `SomeType` is the type which contains the call to `GetExecutingAssembly`.

Let's see how it turns out.

**Comment by rpgmaker** [comment]
Thanks for the effort. GetTypeInfo does not exist either, my guess is that it is not available and I could just use GetType for it.

Thanks

**Comment by rpgmaker** [comment]
I looked into it, GetTypeInfo does exist contrary to what i said previously. But it still does not support required method such as GetUnderlyingEnumType, GetMember, MembersType, and e.t.c

**Comment by NimaAra** [comment]
I am still working on this...

It seems _netstandard1.6_ is the lowest we can go as there are a few stuff that would be missing and I am struggling to figure out how to replace them.

Still investigating...

**Comment by rpgmaker** [comment]
Should I close this?

**Comment by NimaAra** [comment]
Closed :-)
